friendly/src/maria/friendly/messages.cljs

```diff
@@ -122,40 +122,49 @@
 (defn bad-types [info]
   (map type-to-name (remove (partial = 'number) (:types info))))
 
-(defmethod ana/error-message :fn-arity
-  [type info]
-  (str "The function `"
-       (or (:ctor info)
-           (:name info))
-       "` needs "
-       (if (= 0 (-> info :argc))   ;; TODO get arity from meta
-         "more"
-         "a different number of")
-       " arguments."))
-
-(defmethod ana/error-message :invalid-arithmetic
-  [type info]
-  (str "The arithmetic operator `"
-       (name (:js-op info))
-       "` can't be used on non-numbers, like "
-       (humanize-sequence (bad-types info)) "."))
-
-(defmethod ana/error-message :undeclared-var
-  [type {missing-name :suffix
-         :keys [macro-present?]}]
-           (if macro-present?
-             (str "`" missing-name "` is a macro, which can only be used in the first position of a list. A macro doesn't have a value on its own, so it doesn't make sense in this position.")
-             (str "`" missing-name "` hasn't been defined! Perhaps there is a misspelling, or this expression depends on a name that has not yet been evaluated?")))
-
-(defmethod ana/error-message :overload-arity
-  [type info]
-  "This is a 'multiple-arity' function, which has more than one expression accepting same number of arguments.")
+(def analyzer-messages
+  {:fn-arity
+   (fn [type info]
```

Comment:
Just a comment, not a request for change: I recently became a fan of the naming convention of a `_` prefix for arguments one wants to keep named, but that one doesn't use. So this could be, `(fn [_type info]`.f

Reply:
sounds like a useful convention to me